### ISSUE TITLE
CMD API Help needed

### ISSUE THREAD

**Issue by merceyz** [hughperkins/DeepCL#74]
As "requested" in issue #73 i'm moving my question part here

I'm trying to use the CMD API to have a network learn to label some 96x96x3 (RGB) images.

I have a folder structure where each folder for each label has a bunch of images
![dceeb1b8098bbbec435598086613b460](https://cloud.githubusercontent.com/assets/3842800/17055764/48e24804-500f-11e6-901f-c7248bfd87e4.png)
![147bc888df3ea38578d239e6525beb8e](https://cloud.githubusercontent.com/assets/3842800/17055765/4a534292-500f-11e6-9bdf-359d5911a94c.png)
(water)

I opened cmd and ran the following commands:

1. cd "path to folder here"
2. echo off
3. cls
4. call .\bin\activate.bat
5. deepcl_unittests

This resulted in the following output
![970fa81b05292c65274d5321acb3b80c](https://cloud.githubusercontent.com/assets/3842800/17055828/b374dd9e-500f-11e6-922d-042e342f50e5.png)

Now i'm kind of stuck, I was reading on [this page](https://github.com/hughperkins/DeepCL/blob/master/doc/Commandline.md) at the "File types" section there is a line saying "See Loaders for information on available loaders" which doesn't work, the link leads to an 404 page

So my question/issue is: How do I give it data to train/test on?

**Comment by hughperkins** [comment]
Ok, so `deepcl_unittests` is for running automated unit tests, to check stuff is working ok.  To run training you'll want to use `deepcl_train`.

**Comment by merceyz** [comment]
Yeah i figured as much, but how do i present it with my data?

**Comment by hughperkins** [comment]
Ah, good point .Link should work now https://github.com/hughperkins/DeepCL/blob/master/doc/Loaders.md

**Comment by merceyz** [comment]
That jpeg loader will use RGB right or do i have to specify something for that?

**Comment by hughperkins** [comment]
Oh, good question.  I *think* you just need to put `planes=3` in the manifest, and then it should work.  Can you try ,and see if this works ok?

**Comment by hughperkins** [comment]
(I *think* the only requirements on the images are:
- all same size/geometry
- all square, ie width and height are the same
)

**Comment by merceyz** [comment]
![42017794fd46d166352604a8857c98b2](https://cloud.githubusercontent.com/assets/3842800/17075583/1c48dc02-5098-11e6-8c73-6600e2742d97.png)

![b891055c923e4c5ec2a0099ed7186e67](https://cloud.githubusercontent.com/assets/3842800/17075586/30386476-5098-11e6-8303-75ff875bce3a.png)

![4b108e9f3211e10d5a9b6f76dc9380e9](https://cloud.githubusercontent.com/assets/3842800/17075596/62e578b4-5098-11e6-965a-7dec7ecece35.png)

All requirements are met

**Comment by hughperkins** [comment]
Works ok?

**Comment by merceyz** [comment]
"Something went wrong: can't open *file name here"

**Comment by hughperkins** [comment]
Does it really say '*file name here'? or does that have an actual filepath?

**Comment by merceyz** [comment]
If you look at the first screenshot you can see the actual path

**Comment by hughperkins** [comment]
ah...

**Comment by hughperkins** [comment]
ok, can you try removing the leading `/`, ie `0/620cc66f-f919-...etc..`

**Comment by merceyz** [comment]
Same error

**Comment by hughperkins** [comment]
Looks like the filepath is relative to the current directory, not to the datadir.  That's probably not really intuitive, or correct, but I think that's how it works currently. Can you try using either absolute paths, or paths relative to the current directory?  (Might need to add `TestFolder/` to the start of each path?)

**Comment by merceyz** [comment]
That fixed it and it puked out a bunch of stuffs

![a36d961b362d84af1b8acfa950607e39](https://cloud.githubusercontent.com/assets/3842800/17075703/5f83c7c6-509c-11e6-9f3f-7310dd01e445.png)

If i'm not mistaken something is wrong tho as "training loss" says "-1.#IND" and all values are the same on each epoch finished

**Comment by hughperkins** [comment]
ah yeah, that training loss means someting went wrong...

**Comment by hughperkins** [comment]
maybe try a smaller learning rate?

**Comment by merceyz** [comment]
There we go, loss started at ~800 going down more and more

Accuracy on both the training and test keeps climbing so it appears to be working :+1: 

It got to epoch 12 and "clblas teardown" showed up, i assume that's it's built in "limit"

**Comment by hughperkins** [comment]
> Accuracy on both the training and test keeps climbing so it appears to be working

Cool :-)

> It got to epoch 12 and "clblas teardown" showed up, i assume that's it's built in "limit"

By default, it trains 12 epochs :-P  You can change that using `numepochs=100` (or however many epochs yo uwant)

**Comment by merceyz** [comment]
Awesome, thanks.

Suggesting a fix to this:
> Looks like the filepath is relative to the current directory, not to the datadir. That's probably not really intuitive, or correct, but I think that's how it works currently. Can you try using either absolute paths, or paths relative to the current directory? (Might need to add TestFolder/ to the start of each path?)

Otherwise i have no more questions for now so you can close this issue if you like (or keep it open as a reminder of the quote above)

**Comment by hughperkins** [comment]
I'll close this, since it is more general, but you could open an issue specifically for the filepath perhaps?

**Comment by hughperkins** [comment]
(for info, I've fixed the issue with relative path, in the sourcecode.  Question: are you using a prebuilt release, or building from source?)

**Comment by merceyz** [comment]
Prebuilt.
http://deepcl.hughperkins.com/Downloads/
http://deepcl.hughperkins.com/Downloads/deepcl-win64-v8.3.1.zip
